### PR TITLE
Remove dependence on Enumerable utils

### PR DESCRIPTION
**Pull request by paddyobrien**

### CODE REVIEW

File: ember-model.js [hunk 1]
```diff
@@ -340,29 +340,29 @@ Ember.ManyArray = Ember.RecordArray.extend({
...
-    added = Ember.EnumerableUtils.map(added, function(record) {
+    added = added.map(function(record) {
       return record._reference;
     }, this);
 
     this._super(index, removed, added);
   },
 
-  _contentWillChange: function() {
+  _contentDidChange: function() {
```
Comment (GavinJoyce) [inline ember-model.js:350]: Are there tests covering these functions? If not, perhaps we could add them?

Comment (GavinJoyce) [inline ember-model.js:350]: Are there no tests? :scream:

Comment (paddyobrien) [inline ember-model.js:350]: Heh, `package/tests` I had the same thought awhile ago. I'll dig into the tests a bit now and confirm there's coverage here.

Comment (paddyobrien) [inline ember-model.js:350]: Also I should say, these changes were already in upstream, they'd just never been built.

Comment (GavinJoyce) [inline ember-model.js]: :+1: so

Comment (GavinJoyce) [inline ember-model.js:350]: Is there is an open PR on ember-model for this or that they have already been merged? If merged, any reason why we didn't branch off master?

Comment (paddyobrien) [inline ember-model.js:350]: They've been merged. We did branch off master.

They were merged without running `grunt release` so the changes are in the `packages` directory but not in the concated release file that we actually use through bower.

Comment (GavinJoyce) [inline ember-model.js:350]: Thanks. 

:+1:

**Pull request merged**